### PR TITLE
Update documentation for CMath library

### PR DESCRIPTION
**Pull request by davydovanton**
Hello,
I was [created issue](https://bugs.ruby-lang.org/issues/11162) in redmine, but I think that here this changes will be faster to merge :sweat:

### CODE REVIEW

**Comment by gogotanaka** [comment]
@davydovanton Hi! Good proposal.
The things I have to say are:
1. We should mention complex number at `Complex` rather than `CMath`.
2. `3 + 2i`(Ruby>=2.1) can also represent complex number.

thanks!

**Comment by davydovanton** [comment]
@gogotanaka wow! thanks for reply! :star2: 
1. I think I can update documentation for `Complex` class too.
2. wow, I don't know about this, thanks!
3. okay, I will fix this later

**Comment by davydovanton** [comment]
@gogotanaka I change example to `1 + 1i`. Also I think that I need update `Complex` documentation in  the next PR. What do you think, it's normal for you?

File: lib/cmath.rb [hunk 1]
```diff
@@ -1,16 +1,49 @@
...
+# functions for complex numbers. The functions in this module accept
+# integers, floating-point numbers or complex numbers as arguments. 
+#
+# Note that the selection of functions is similar, but not identical,
+# to that in module math. The reason for having two modules is that
+# some users aren’t interested in complex numbers, and perhaps don’t
+# even know what they are. They would rather have Math.sqrt(-1) raise
+# an exception than return a complex number.
+#
+# == Brief overview of complex numbers
```
Comment (gogotanaka) [inline lib/cmath.rb]: @davydovanton Thank you for updating! How about having this section at `Complex` and putting link to `Complex` here?
sorry I just wanna avoid duplicate contents.

**Comment by davydovanton** [comment]
@gogotanaka so, I did replaced `Brief overview` to `Complex` class, it's normal for you?

**Comment by DJGHOSTMACHINE** [comment]
I will let you know

**Comment by davydovanton** [comment]
ok

**Comment by gogotanaka** [comment]
@davydovanton Merged :star: thanks!
note: Since the documentation you've add to `complex.c` is duplication of existing one, I don't commit it for now.
But if you wanna improve it more, please feel free to open another ticket :D

**Comment by davydovanton** [comment]
@gogotanaka thanks for review :star2:

**Pull request closed without merging**